core/unit_test/TestMathematicalFunctions.hpp

```diff
@@ -728,6 +759,48 @@ void do_test_math_binary_function(Arg1 arg1, Arg2 arg2) {
       (TestMathBinaryFunction<Space, Func, Arg1, Arg2>(arg1, arg2), 0)...};
 }
 
+template <class Space, class Func, class Arg1, class Arg2,
+          class Ret = math_binary_function_return_type_t<Arg1, Arg2>>
+struct TestMathTernaryIntPtrFunction : FloatingPointComparison {
+  Arg1 val1_;
+  Arg2 val2_;
+  int valptr_;
+  Ret res_;
+  TestMathTernaryIntPtrFunction(Arg1 val1, Arg2 val2)
+      : val1_(val1), val2_(val2), res_(Func::eval_std(val1, val2, &valptr_)) {
+    run();
+  }
+  void run() {
+    int errors = 0;
+    Kokkos::parallel_reduce(Kokkos::RangePolicy<Space>(0, 1), *this, errors);
+    ASSERT_EQ(errors, 0) << "Failed check no error for "
+                         << math_function_name<Func>::name << "("
+                         << type_helper<Arg1>::name() << ", "
+                         << type_helper<Arg2>::name() << ")";
+  }
+  KOKKOS_FUNCTION void operator()(int, int& e) const {
+    int valptr;
+    bool ar_1 =
+        compare(Func::eval(val1_, val2_, &valptr), res_, Func::ulp_factor());
+    bool ar_2 = compare(valptr_, valptr, Func::ulp_factor());
```

Comment:
I am confused about that comparison.  What does it do?
Why are we not doing `*valptr_ == *valptr`

Reply:
We could use `==`, we probably don't need to have a tolerance. 

We don't want to deference `valptr` or `valptr_`. There are integers.

Reply:
Maybe let's drop the `*prr` suffix so that inattentive readers (aka me) don't get confused